packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -805,7 +805,7 @@ public void run() {
                   ? MetadataChanges.INCLUDE
                   : MetadataChanges.EXCLUDE;
           listenerRegistrations.put(
-              handle, getQuery(arguments).addSnapshotListener(metadataChanges, observer));
+              handle, getQuery(arguments).addSnapshotListener(activity, metadataChanges, observer));
```

Comment:
`activity` will be `null` when the App is run in background only.

It really should be annotated with `@Nullable`.